stdlib/public/core/StringGuts.swift

```diff
@@ -219,26 +219,40 @@ extension _StringGuts {
 // C String interop
 extension _StringGuts {
   @inlinable @inline(__always) // fast-path: already C-string compatible
-  internal func withCString<Result>(
-    _ body: (UnsafePointer<Int8>) throws -> Result
-  ) rethrows -> Result {
+  internal func withCString<Success, Failure: Error>(
+    _ body: (UnsafePointer<Int8>) throws(Failure) -> Success
+  ) throws(Failure) -> Success {
     if _slowPath(!_object.isFastZeroTerminated) {
       return try unsafe _slowWithCString(body)
     }
 
-    return try unsafe self.withFastCChar {
-      return try unsafe body($0.baseAddress._unsafelyUnwrappedUnchecked)
+    return try unsafe self.withFastCChar { (pointer) throws(Failure) in
+      return try unsafe body(pointer.baseAddress._unsafelyUnwrappedUnchecked)
     }
   }
 
+  @abi(
+    func _slowWithCString<Result>(
+      _ body: (UnsafePointer<Int8>) throws -> Result
+    ) rethrows -> Result
+  )
   @inline(never) // slow-path
   @usableFromInline
-  internal func _slowWithCString<Result>(
+  internal func __rethrows_slowWithCString<Result>(
     _ body: (UnsafePointer<Int8>) throws -> Result
-  ) rethrows -> Result {
+  ) throws -> Result {
+    return try unsafe self._slowWithCString(body)
+  }
+
+  @inline(never) // slow-path
+  @usableFromInline
+  @_alwaysEmitIntoClient
```

Comment:
`@usableFromInline @_alwaysEmitIntoClient internal` does not seem like a valid combination of attributes.

```suggestion
  @_alwaysEmitIntoClient
```

Reply:
Not valid in what sense? Could that be diagnosed by the compiler a trigger error diagnostics? It clearly did not do that and compiled successfully.

Reply:
Not valid in that it doesn't make sense. It could be diagnosed by the compiler. `@usableFromInline` says that the declaration of this entity is available externally (aka it is ABI), while `@_aEIC` says the definition of this entity is always available and gets copied into client binaries (aka it's not ABI).